transmogrifier/sources/transformer.py

```diff
@@ -248,7 +248,11 @@ def _transform(
             title=self.get_valid_title(source_record),
         )
         for field_name, field_method in self.get_optional_field_methods():
-            setattr(timdex_record, field_name, field_method(source_record))
+            try:
+                setattr(timdex_record, field_name, field_method(source_record))
+            except:  # noqa: E722
```

Comment:
Not sure if we want to narrow this bare exception block and I'm also not sure the best way of testing that an exception triggers this behavior, I did some experiments with the already-imported `Datacite` class but had no luck

Reply:
What is the result of this `try/except` block when the `field_method(source_record)` raises a `SkippedRecordEvent` (e.g., `Datacite.get_content_type`)?

Reply:
I believe it would catch `SkippedRecordEvent`, log it and raise it again, does that seem right @ghukill? It's a double dip on the exception in that case but perhaps not a big problem if it is catching all other exceptions and ensuring the record is skipped rather than tanking the whole run

Reply:
Hmm, is there a particular reason why we don't do the following:

```
try: 
   setattr(timdex_record, field_name, field_method(source_record))
except Exception as exc:
   raise SkippedRecordEvent from exc
```

Here is the output I'm seeing:

```
try:
   raise SkippedRecordEvent("Blah")
except Exception as exc:
   raise SkippedRecordEvent from exc

# Output 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
transmogrifier.exceptions.SkippedRecordEvent: Blah

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
transmogrifier.exceptions.SkippedRecordEvent: None
```

Reply:
Another option is shown below, but it feels incorrect to say "another" exception occurred. 🤔 

```
try:
    raise SkippedRecordEvent("Blah")
except Exception as exc:
    raise SkippedRecordEvent(exc)

# Output
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
transmogrifier.exceptions.SkippedRecordEvent: Blah

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
transmogrifier.exceptions.SkippedRecordEvent: Blah
```

Reply:
Maybe this is the solution (pending `ruff` saying "nope!" 😅 )

```
try:
   raise SkippedRecordEvent("useful error message")
except Exception as exc:
   raise SkippedRecordEvent(exc) from exc
```

Output: 

```
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
transmogrifier.exceptions.SkippedRecordEvent: useful error message

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
transmogrifier.exceptions.SkippedRecordEvent: useful error message
```